containerz/containerz.proto

```diff
@@ -64,6 +80,24 @@ service Containerz {
   // (i.e. kill) a container.
   rpc StopContainer(StopContainerRequest) returns (StopContainerResponse) {}
 
+  // UpdateContainer updates a running container to the image specified in the
+  // request. By default the operation is synchronous which means that the
+  // request will only return once the container has either been successfully
+  // updated or the update has failed. If the client requests an asynchronous
+  // update then the server must perform all validations (e.g. does the
+  // requested image exist on the system or does the instance name exist) and
+  // return to the client and the update happens asynchronously. If the update
+  // fails, the server must restore the previous version of the container. It
```

Comment:
It's not quite clear to me what the order of operations of this RPC is expected to me. Is it make-before-break or break-before-make? I assume that `UpdateContainer()` being called will mean that we shut down the existing instance and then start a new using the new image. I assume that this'll be required, since otherwise resources (e.g., TCP ports) that are needed by the new instance may still be bound to the old instance.

Rollback then (as Sergey says above) is really "start a new container using the old image and config", I presume? Can we clarify?

Reply:
I explicitely mentioned that this is break-before-make. 

For rollback, the server can either restart the stopped container or it can start a new container with the old image. Both would be equivalent. I added text making this explicit.

Reply:
As per discussion -- the client should only depend on the process being restarted -- the contents of the filesystem are NOT guaranteed to be restored after a rollback operation.